NuRadioMC/simulation/simulation.py

```diff
@@ -1206,23 +1239,34 @@ def _create_station_output_structure(self, n_showers, n_antennas):
                 sg[parameter_entry['name']] = np.zeros((n_showers, n_antennas, nS, parameter_entry['ndim'])) * np.nan
         return sg
 
-    def _read_input_neutrino_properties(self):
+    def _read_input_particle_properties(self):
+        self._n_interaction = self._fin['n_interaction'][self._shower_index]
         self._event_group_id = self._fin['event_group_ids'][self._shower_index]
-        self._flavor = self._fin['flavors'][self._shower_index]
-        self._energy = self._fin['energies'][self._shower_index]
-        self._inttype = self._fin['interaction_type'][self._shower_index]
-        self._x = self._fin['xx'][self._shower_index]
-        self._y = self._fin['yy'][self._shower_index]
-        self._z = self._fin['zz'][self._shower_index]
+
+        self.input_particle = NuRadioReco.framework.particle.Particle(0)
+        self.input_particle[simp.flavor] = self._fin['flavors'][self._shower_index]
+        self.input_particle[simp.energy] = self._fin['energies'][self._shower_index]
+        self.input_particle[simp.interaction_type] = self._fin['interaction_type'][self._shower_index]
+        self.input_particle[simp.inelasticity] = self._fin['inelasticity'][self._shower_index]
+        self.input_particle[simp.vertex] = np.array([self._fin['xx'][self._shower_index],
+                                                  self._fin['yy'][self._shower_index],
+                                                  self._fin['zz'][self._shower_index]])
+        self.input_particle[simp.zenith] = self._fin['zeniths'][self._shower_index]
+        self.input_particle[simp.azimuth] = self._fin['azimuths'][self._shower_index]
+        self.input_particle[simp.inelasticity] = self._fin['inelasticity'][self._shower_index]
+        self.input_particle[simp.parent_id] = None # primary does not have a parent
```

Comment:
why is it always a primary? Don't you need to consult `n_interations` to see if it is the primary?

Reply:
yes you are right. should test with a proposal file to see it is handled correctly, the n_interactions is a good suggestion